src/Files.App/Helpers/Application/AppLifecycleHelper.cs

```diff
@@ -259,6 +259,12 @@ public static void HandleAppUnhandledException(Exception? ex, bool showToastNoti
 
 			if (ex is not null)
 			{
+				ex.ToExceptionless()
+					.SetReferenceId(Guid.NewGuid().ToString())
+					.Submit();
+
+				await ExceptionlessClient.Default.ProcessQueueAsync();
```

Comment:
The method returns when it hit the first `await` and causes the app closing.
You need to use `ExceptionlessClient.Default.ProcessQueueAsync().ConfigureAwait(false).GetAwaiter().GetResult()` instead, also make this method non-async.

Reply:
Should I set this here or when I first configure the client?